sdk/python/feast/feature_store.py

```diff
@@ -805,28 +825,61 @@ def get_online_features(
         for entity in entities:
             entity_name_to_join_key_map[entity.name] = entity.join_key
 
+        needed_request_data_features = self._get_needed_request_data_features(
+            grouped_odfv_refs
+        )
+
         join_key_rows = []
+        request_data_features: Dict[str, List[Any]] = {}
+        # Entity rows may be either entities or request data.
         for row in entity_rows:
             join_key_row = {}
             for entity_name, entity_value in row.items():
+                # Found request data
+                if entity_name in needed_request_data_features:
+                    if entity_name not in request_data_features:
+                        request_data_features[entity_name] = []
+                    request_data_features[entity_name].append(entity_value)
+                    continue
                 try:
                     join_key = entity_name_to_join_key_map[entity_name]
                 except KeyError:
                     raise EntityNotFoundException(entity_name, self.project)
                 join_key_row[join_key] = entity_value
                 if entityless_case:
                     join_key_row[DUMMY_ENTITY_ID] = DUMMY_ENTITY_VAL
-            join_key_rows.append(join_key_row)
+            if len(join_key_row) > 0:
+                # May be empty if this entity row was request data
+                join_key_rows.append(join_key_row)
+
+        if len(needed_request_data_features) != len(request_data_features.keys()):
+            raise RequestDataNotFoundInEntityRowsException(
+                feature_names=needed_request_data_features
+            )
```

Comment:
Can we add a test that asserts this exception is raised correctly?

Reply:
Done